main.tf

```diff
@@ -13,9 +13,16 @@ locals {
 
   get_cluster_data = local.enabled ? (local.need_cluster_kubernetes_version || local.need_bootstrap || local.need_ssh_access_sg || length(var.associated_security_group_ids) > 0) : false
 
+  taint_effect_map = {
```

Comment:
why this is necessary?
why the use can't use `NoSchedule` instead of `NO_SCHEDULE`?

Reply:
Because we need `NO_SCHEDULE` in one place and `NoSchedule` in another based on the same input which is already documented to require `NO_SCHEDULE`.